tensorvis/tensorvisbase/src/datastructures/tensorfield3d.cpp

```diff
@@ -178,6 +178,46 @@ TensorField3D::TensorField3D(const size_t x, const size_t y, const size_t z,
     computeDataMaps();
 }
 
+TensorField3D::TensorField3D(size3_t dimensions, const double *data, const dvec3 &extent,
+                             double sliceCoord)
+    : dimensions_(dimensions)
+    , extent_(extent)
+    , indexMapper_(dimensions)
+    , size_(dimensions.x * dimensions.y * dimensions.z)
+    , rank_(2)
+    , dimensionality_(3)
+    , offset_(dvec3(0.0)) {
+    tensors_.resize(size_);
+    std::memcpy(tensors_.data(), data, size_ * sizeof(double) * 9);
+
+    computeEigenValuesAndEigenVectors();
+    computeNormalizedScreenCoordinates(sliceCoord);
+    computeDataMaps();
+}
+
+TensorField3D::TensorField3D(size3_t dimensions, const float *data, const dvec3 &extent,
+                             double sliceCoord)
+    : dimensions_(dimensions)
+    , extent_(extent)
+    , indexMapper_(dimensions)
+    , size_(dimensions.x * dimensions.y * dimensions.z)
+    , rank_(2)
+    , dimensionality_(3)
+    , offset_(dvec3(0.0)) {
+    // Convert float to double
+    std::vector<float> floatArray;
```

Comment:
Why not copy the data directly into the tensors?
```c++
tensors_.resize(size_);
for (size_t i = 0; i<size_; ++i) {
    std::copy(data + i * 9, data + (i + 1) * 9, glm::value_ptr(tensors_[i]));
}
```

Should also take care of float to double conversions.

Reply:
Done.